### PR TITLE
Fix "index out of range" runtime error in repo_list tests

### PR DESCRIPTION
**Pull request by Morlinest**
Fix runtime errors in specific conditions.

### CODE REVIEW

File: models/repo_list_test.go [hunk 1]
```diff
@@ -23,7 +23,9 @@ func TestSearchRepositoryByName(t *testing.T) {
 
 	assert.NotNil(t, repos)
 	assert.NoError(t, err)
-	assert.Equal(t, "test_repo_12", repos[0].Name)
+	if len(repos) > 0 {
```
Comment (bkcsoft) [inline models/repo_list_test.go]:
```go
if assert.NotEmpty(repos) {
  assert.Equal(t, "test_repo_12", repos[0].Name)
}
```

File: models/repo_list_test.go [hunk 2]
```diff
@@ -48,7 +50,9 @@ func TestSearchRepositoryByName(t *testing.T) {
 
 	assert.NotNil(t, repos)
 	assert.NoError(t, err)
-	assert.Equal(t, "test_repo_13", repos[0].Name)
+	if len(repos) > 0 {
```
Comment (bkcsoft) [inline models/repo_list_test.go]: Same as above

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: models/repo_list_test.go (hunk 1 above)
Comment (Morlinest) [reply models/repo_list_test.go]: You are right, my first tests :D

**Comment by strk** [comment]
it's a test, shouldn't you know what to expect (repo being or not being empty) ?
I'd think if a repo is unexpectedly empty there is some other problem (the test will never run?)

**Comment by Morlinest** [comment]
@strk Sure, you should know what to expect. But that doesn't mean, you can write tests that can ends by `runtime error`. I think event creator of tests will have problems to know what that mean without looking to code. If someone else wants to run tests, he don't know what that runtime error means at all without looking to code. I don't see any reason to write any code that can contains bugs, "it is only test" is not excuse.

I think this:
image

is better than:
image

**Comment by Morlinest** [comment]
Also, `panic: runtime error` means end of any other tests = you will be missing tests after that line.

**Comment by strk** [comment]
Yep, an error about unexpected empty is great. I didn't notice the assert there.
LGTM

**Comment by lafriks** [comment]
LGTM
@bkcsoft need your approval

File: models/repo_list_test.go [hunk 3]
```diff
@@ -23,7 +23,9 @@ func TestSearchRepositoryByName(t *testing.T) {
 
 	assert.NotNil(t, repos)
 	assert.NoError(t, err)
-	assert.Equal(t, "test_repo_12", repos[0].Name)
+	if assert.NotEmpty(t, repos) {
```
Comment (lunny) [inline models/repo_list_test.go]: ~No. You should assert how many repos are there.~

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
But it seems it's no need to do that?

**Review by bkcsoft — APPROVED** [review]

File: models/repo_list_test.go (hunk 3 above)
Comment (bkcsoft) [reply models/repo_list_test.go]: We only need to know that there's more than 0 repos. (since we're accessing `repos[0]`)

**Review by lunny — APPROVED** [review]

**Pull request merged**